onnxruntime/core/providers/openvino/openvino_graph.cc

```diff
@@ -0,0 +1,464 @@
+  // (which also contains initializers).
+  std::map<std::string, int> inputdef_index_map;
+  auto input_defs = fused_node_->InputDefs();
+  int i = 0;
+  for (auto idef : input_defs) {
+    inputdef_index_map.insert({idef->Name(), i});
+    i++;
+  }
+
+  auto inputs = fused_node_->GetFunctionBody()->Body().GetInputs();
+  for (auto input : inputs) {
+    if (inputdef_index_map.find(input->Name()) == inputdef_index_map.end()) {
+      throw "Input not found in the input defs list";
+    }
+
+    int index = inputdef_index_map[input->Name()];
+    input_indexes_.push_back(index);
+  }
+
+  // Create hardware agnostic OpenVINO network representation
+  openvino_network_ = BuildOpenVINONetworkWithMO();
+
+  // Create hardware specific OpenVINO network representation
+  infer_requests_ = GetExecutableHandle(openvino_network_, device_id_, precision_);
+}
+
+std::vector<std::string> OpenVINOGraph::GetEnvLdLibraryPath() {
+  std::string plugin_path = std::getenv("LD_LIBRARY_PATH");
+  std::vector<std::string> paths;
+  std::string token;
+  std::istringstream tokenStream(plugin_path);
+  char delimiter = ':';
+
+  while (std::getline(tokenStream, token, delimiter)) {
+    paths.push_back(token);
+  }
+  return paths;
+}
+
+void OpenVINOGraph::ConvertONNXModelToOpenVINOIR(std::string& onnx_model,
+    std::string& openvino_xml, std::string& openvino_bin, bool precision_fp32) {
+
+  Py_Initialize();
+  if (!Py_IsInitialized()) {
+    throw "Python environment initialization failure";
+  }
+
+  PyObject* pModule = NULL;
+  pModule = PyImport_ImportModule("openvino_mo");
+  if (pModule == NULL) {
+    throw "Python module import failure";
+  }
+
+  PyObject* pFunc = NULL;
+  if (precision_fp32) {
+    pFunc = PyObject_GetAttrString(pModule, "convert_fp32");
+  } else {
+    pFunc = PyObject_GetAttrString(pModule, "convert_fp16");
+  }
+  if (pFunc == NULL || !PyCallable_Check(pFunc)) {
+    throw "Python module function check failure";
+  }
+
+  // Prepare ModelProto Input to Python
+  PyObject* pFileName = PyByteArray_FromStringAndSize(
+      onnx_model.c_str(), onnx_model.size());
+  PyObject* pArgs = PyTuple_New(1);
+  PyTuple_SetItem(pArgs, 0, pFileName);
+
+  PyObject* pOutputTuple = NULL;
+
+  // Call the Python function
+  pOutputTuple = PyObject_CallObject(pFunc, pArgs);
+
+  if (pOutputTuple == NULL || !PyTuple_CheckExact(pOutputTuple)) {
+    throw "Python function call failure";
+  }
+
+  // Retrieve the weights byte array
+  PyObject* pArg1 = PyTuple_GetItem(pOutputTuple, 0);
+  PyObject* pWeights = PyByteArray_FromObject(pArg1);
+  const char* weights_bytes = PyByteArray_AsString(pWeights);
+  unsigned long weights_size = PyByteArray_Size(pWeights);
+  std::string weights_string(weights_bytes, weights_size);
+  openvino_bin = weights_string;
+
+  // Retrieve the xml string
+  PyObject* pArg2 = PyTuple_GetItem(pOutputTuple, 1);
+  PyObject* pXML = PyObject_Repr(pArg2);
+  openvino_xml = PyUnicode_AsUTF8(pXML);
+
+  //    Py_DECREF(pXML);
```

Comment:
why commented out?

Reply:
Embedded python interpreter currently throws an error if python modules are being loaded more than once from the same process (after a PyFinalize and re-PyInitialize). This causes the unit test framework to fail because it creates multiple Onnxruntime sessions from the same process that causes multiple python module inclusions. The above code is commented out temporarily while we're debugging this crash. Will re-enable these statements back along with the fix.

Reply:
Now re-enabled code to decrement reference counts of pyObjects.